src/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ExecutionEnvironmentImplementation.MappingTables.cs

```diff
@@ -343,7 +343,16 @@ public unsafe sealed override bool TryGetConstructedGenericTypeForComponents(Run
 
             TypeInfo[] typeArguments = new TypeInfo[genericTypeArgumentHandles.Length];
             for (int i = 0; i < genericTypeArgumentHandles.Length; i++)
+            {
+                // Early out if one of the arguments is a generic definition.
+                // The reflection stack will use this to construct a Type that doesn't have a type handle.
+                // Note: this is different from the validation we do in EnsureSatisfiesClassConstraints because this
+                // should not throw.
+                if (RuntimeAugments.IsGenericTypeDefinition(genericTypeArgumentHandles[i]))
```

Comment:
Good point. Seems like that is not the right place to throw the exception either. The CLR allows `typeof(List<>).MakeArrayType();` just fine.

We don't know the intent of the higher level caller at this point, so returning `false` is what the protocol would dictate. It's up to the caller to decide what to do with that.

Cc @AtsushiKan